src/Expander.hs

```diff
@@ -562,6 +628,39 @@ initializeKernel = do
       [ ( "oops"
         , \ _ stx -> throwError (InternalError $ "oops" ++ show stx)
         )
+      , ( "the"
+        , \ dest stx -> do
+            Stx _ _ (_, ty, expr) <- mustHaveEntries stx
+            tyDest <- scheduleType ty
+            -- TODO add type to elaborated program? Or not?
```

Comment:
It would make sense to me to have the output of expansion+typechecking be something like
```haskell
newtype TypedCore = TypedCore { getTypedCore :: (Type, CoreF TypedCore) }
```
for easy analysis (e.g. thinking of an optimization pass), and then have the input to evaluation be `Core`. Then after expansion but before evaluation we could
```haskell
eraseTypes :: TypedCore -> Core
```